### ISSUE TITLE
Mangled line endings from cms -sign

### ISSUE THREAD

**Issue by stevemit** [openssl/openssl#7763]
In OpenSSL 1.1.1a, the following command produces odd line endings (\r\r\n) in the output file `signed.msg`.
```
openssl cms -sign -in message.txt -out signed.msg -signer sigkey.cer -inkey sigkey.pem -certfile sig-ca.cer
```
If `message.txt` contains document A:
```
Content-Type: text/plain\r\n
\r\n
Hello, world!\r\n
```
the digested and signed document is exactly `message.txt`, but the output file `signed.msg` includes document B:
```
Content-Type: text/plain\r\r\n
\r\r\n
Hello, world!\r\r\n
```
If we verify the signature with
```
openssl cms -verify -in signed.msg -out signedtext.txt -signer sigkey.cer -CAfile sig-ta.cer
```
the verification of document A is successful, but the output file `signedtext.txt` contains document B.
If we replace document B with document A in `signed.msg`, the result is the same – the verification of document A is successful, but the tool outputs document B.

Document A is already an S/MIME canonical document (\r\n line endings), while document B is not.

**Comment by kroeckx** [comment]
Try using the -binary option

**Comment by kroeckx** [comment]
Or the -crlfeol option

**Comment by stevemit** [comment]
I am testing on Windows 10 in Command Prompt, using a binary build from [bintray.com](https://bintray.com/vszakats/generic/openssl/1.1.1a).  Maybe Linux gives different results, but these commands are broken on Windows.  However, the actual signed document is correct, and signature validation is correct.

It's not clear why one would add the `-binary` option to process a text file.  If so, when would one not use the `-binary` option?  The `-crlfeol` option is documented as producing \r\n instead of \n in the output files, which seems irrelevant to Windows.

Adding the `-binary` option to the sign command has no effect -- the `signed.msg` output is still document B.  Adding the `-binary` option to the verify command produces document A as `signedtext.txt`.

Adding the `-crlfeol` option to the sign command produces \r\r\n line endings not only in the signed document MIME part but throughout the `signed.msg` output file.  Adding the `-crlfeol` option to the verify command has no effect -- the `signedtext.txt` file is still document B.

**Comment by gboor** [comment]
I just ran into this on Windows 10 using the win64 binary build of OpenSSL. I can confirm this has still not been fixed and no amount of options fix it... It seems that all input \n AND \r\n are turned into \r\r\n in the output, which is not very useful.

**Comment by t8m** [comment]
This might be fixed in OpenSSL-3.0 as there was fixes related done there.
